lib/pacote-opts.js

```diff
@@ -0,0 +1,135 @@
+'use strict'
+
+const Buffer = require('safe-buffer').Buffer
+
+const crypto = require('crypto')
+const path = require('path')
+
+let effectiveOwner
+
+const npmSession = crypto.randomBytes(8).toString('hex')
+
+module.exports = pacoteOpts
+function pacoteOpts (npmOpts, moreOpts) {
+  const ownerStats = calculateOwner()
+  const opts = {
+    cache: path.join(npmOpts['cache'], '_cacache'),
+    ca: npmOpts['ca'],
+    cert: npmOpts['cert'],
+    git: npmOpts['git'],
+    key: npmOpts['key'],
+    localAddress: npmOpts['local-address'],
+    loglevel: npmOpts['loglevel'],
+    maxSockets: +npmOpts['maxsockets'],
+    npmSession: npmSession,
+    offline: npmOpts['offline'],
+    projectScope: getProjectScope((npmOpts.rootPkg || moreOpts.rootPkg).name),
+    proxy: npmOpts['https-proxy'] || npmOpts['proxy'],
+    refer: 'cipm',
+    registry: npmOpts['registry'],
+    retry: {
+      retries: npmOpts['fetch-retries'],
+      factor: npmOpts['fetch-retry-factor'],
+      minTimeout: npmOpts['fetch-retry-mintimeout'],
+      maxTimeout: npmOpts['fetch-retry-maxtimeout']
+    },
+    strictSSL: npmOpts['strict-ssl'],
+    userAgent: npmOpts['user-agent'],
+
+    dmode: parseInt('0777', 8) & (~npmOpts['umask']),
```

Comment:
Seems like this and fmode can move into pacote itself, considering its derived from umask

Reply:
I'm not sure pacote itself even needs to be setting these, since they are only used when a tarball doesn't contain mode details for file(s) it contains, which it should. This value is _also_ the same as the default `dmode` in `tar`. Same with `fmode` below.